### ISSUE TITLE
cannot run manage.py test with postgres DB

### ISSUE THREAD

**Issue by bb4L** [tolomea/django-data-browser#23]
when trying to run python manage.py test with postgres it results in a `django.db.utils.InternalError: current transaction is aborted, commands ignored until end of transaction block`

(with sqlite it would work, but since tests are part of a CI and the deployed instance will use postgres i would prefer to run the tests with a postgres)

here the detailed trace:
```
Creating test database for alias 'default'...
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
psycopg2.errors.InFailedSqlTransaction: current transaction is aborted, commands ignored until end of transaction block


The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "manage.py", line 21, in <module>
    main()
  File "manage.py", line 17, in main
    execute_from_command_line(sys.argv)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 381, in execute_from_command_line
    utility.execute()
  File "/usr/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 375, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/commands/test.py", line 23, in run_from_argv
    super().run_from_argv(argv)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 323, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/commands/test.py", line 53, in handle
    failures = test_runner.run_tests(test_labels)
  File "/usr/local/lib/python3.6/site-packages/django/test/runner.py", line 629, in run_tests
    old_config = self.setup_databases(aliases=databases)
  File "/usr/local/lib/python3.6/site-packages/django/test/runner.py", line 554, in setup_databases
    self.parallel, **kwargs
  File "/usr/local/lib/python3.6/site-packages/django/test/utils.py", line 174, in setup_databases
    serialize=connection.settings_dict.get('TEST', {}).get('SERIALIZE', True),
  File "/usr/local/lib/python3.6/site-packages/django/db/backends/base/creation.py", line 72, in create_test_db
    run_syncdb=True,
  File "/usr/local/lib/python3.6/site-packages/django/core/management/__init__.py", line 148, in call_command
    return command.execute(*args, **defaults)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 364, in execute
    output = self.handle(*args, **options)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/base.py", line 83, in wrapped
    res = handle_func(*args, **kwargs)
  File "/usr/local/lib/python3.6/site-packages/django/core/management/commands/migrate.py", line 234, in handle
    fake_initial=fake_initial,
  File "/usr/local/lib/python3.6/site-packages/django/db/migrations/executor.py", line 117, in migrate
    state = self._migrate_all_forwards(state, plan, full_plan, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.6/site-packages/django/db/migrations/executor.py", line 147, in _migrate_all_forwards
    state = self.apply_migration(state, migration, fake=fake, fake_initial=fake_initial)
  File "/usr/local/lib/python3.6/site-packages/django/db/migrations/executor.py", line 245, in apply_migration
    state = migration.apply(state, schema_editor)
  File "/usr/local/lib/python3.6/site-packages/django/db/migrations/migration.py", line 124, in apply
    operation.database_forwards(self.app_label, schema_editor, old_state, project_state)
  File "/usr/local/lib/python3.6/site-packages/django/db/migrations/operations/special.py", line 190, in database_forwards
    self.code(from_state.apps, schema_editor)
  File "/usr/local/lib/python3.6/site-packages/data_browser/migrations/0009_migrate_saved_views.py", line 10, in forwards
    forwards_0009(View)
  File "/usr/local/lib/python3.6/site-packages/data_browser/migration_helpers.py", line 58, in forwards_0009
    for view in View.objects.all():
  File "/usr/local/lib/python3.6/site-packages/django/db/models/query.py", line 274, in __iter__
    self._fetch_all()
  File "/usr/local/lib/python3.6/site-packages/django/db/models/query.py", line 1242, in _fetch_all
    self._result_cache = list(self._iterable_class(self))
  File "/usr/local/lib/python3.6/site-packages/django/db/models/query.py", line 55, in __iter__
    results = compiler.execute_sql(chunked_fetch=self.chunked_fetch, chunk_size=self.chunk_size)
  File "/usr/local/lib/python3.6/site-packages/django/db/models/sql/compiler.py", line 1142, in execute_sql
    cursor.execute(sql, params)
  File "/usr/local/lib/python3.6/site-packages/raven/contrib/django/client.py", line 127, in execute
    return real_execute(self, sql, params)
  File "/usr/local/lib/python3.6/site-packages/django/db/backends/utils.py", line 67, in execute
    return self._execute_with_wrappers(sql, params, many=False, executor=self._execute)
  File "/usr/local/lib/python3.6/site-packages/django/db/backends/utils.py", line 76, in _execute_with_wrappers
    return executor(sql, params, many, context)
  File "/usr/local/lib/python3.6/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
  File "/usr/local/lib/python3.6/site-packages/django/db/utils.py", line 89, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "/usr/local/lib/python3.6/site-packages/django/db/backends/utils.py", line 84, in _execute
    return self.cursor.execute(sql, params)
django.db.utils.InternalError: current transaction is aborted, commands ignored until end of transaction block  
```

**Comment by tolomea** [comment]
That migration is going to be the bane of my existence.

> i would prefer to run the tests with a postgres

I agree entirely.

My day job also uses Postgres and we aren't having any issues with this. What Postgres and psycopg versions do you have?

I feel like the trace is missing something, do you have any idea what caused the "current transaction is aborted"?

You could try adding ```LOGGING["loggers"]["django.db"] = {"handlers": ["simple"], "level": "DEBUG"}``` to the end of your settings.py (you will need to choose a handler appropriate to your settings) and see if that turns up any additional information.

**Comment by bb4L** [comment]
> My day job also uses Postgres and we aren't having any issues with this. What Postgres and psycopg versions do you have?

postgres 9.6
psycopg2 2.8.6

> I feel like the trace is missing something, do you have any idea what caused the "current transaction is aborted"?

not sure if something is missing... it's all i got

i will try to add the logging and post the output

**Comment by bb4L** [comment]
ok I could "resolve" the issue somehow...

by first applying migrations of some of the apps and then applying the rest

it looks to me that it might be related to having a one to one relationship to the User model

**Comment by tolomea** [comment]
Ah, it might be because the migration calls get_user_model, thinking about
it that doesn't seem right, but I don't know what the correct version is.

Do you have a custom user model?
If so, was is created in your first migration or later?
And does migrating the app with the user model before databrowser fix the
problem?

**Comment by bb4L** [comment]
i don't know if it helps but:
The `View.objects.all()` has no entries

when doing `View.objects.count()` it gives me 0.

and it's of type `__fake__.View`

**Comment by tolomea** [comment]
That I expect. This evening I will make a change so it doesn't even import
the rest of it if you have no View's
That should prevent your problem, but I'm still curious about what exactly
is failing.

Do you have a custom user model?
If so, was it created in your first migration or later?

**Comment by tolomea** [comment]
I've released that as 3.1.1 please let me know if it helps.

**Comment by tolomea** [comment]
and the the universe punished me for not waiting for CI

**Comment by bb4L** [comment]
> I've released that as 3.1.1 please let me know if it helps.

yes it resolves the issue

thank you very much
